src/libfetchers/git.cc

```diff
@@ -590,6 +627,42 @@ struct GitInputScheme : InputScheme
 
         printTalkative("using revision %s of repo '%s'", input.getRev()->gitRev(), actualUrl);
 
+         if (verifyCommit) {
+            // Create ad-hoc allowedSignersFile and populate it with publicKeys
+            auto allowedSignersFile = createTempFile().second;
+            std::string allowedSigners;
+            for (const PublicKey& k : publicKeys) {
+                if (k.type != "ssh-dsa" && k.type != "ssh-ecdsa" && k.type != "ssh-ecdsa-sk"
+                && k.type != "ssh-ed25519" && k.type != "ssh-ed25519-sk" && k.type != "ssh-rsa")
+                    warn("Unknow keytype: %s\n"
+                         "Please use one of ssh-dsa, ssh-ecdsa, ssh-ecdsa-sk, ssh-ed25519, ssh-ed25519-sk, ssh-rsa", k.type);
```

Comment:
the most readable way I found to achive that is 
```
std::vector<std::string> valid_types{"ssh-dsa", "ssh-ecdsa", "ssh-ecdsa-sk", "ssh-ed25519", "ssh-ed25519-sk", "ssh-rsa"};
for (const PublicKey& k : publicKeys) {
    if ((std::find(valid_types.begin(), valid_types.end(), k.type) == valid_types.end()))
        warn("Unknow keytype: %s\n"
             "Please use one of " + boost::algorithm::join(valid_types, ", "), k.type);
    allowedSigners += "* " + k.type + " " + k.key + "\n";
}
```
which feels significantly less readable to me (thogh that might be because I'm not used to cpp code).
 Do you have a different suggestion? Otherwise I'd opt for making the condition one per line.